pkg/cosi/provisioner.go

```diff
@@ -0,0 +1,616 @@
+		HasLogin:          false,
+		S3Access:          true,
+		AllowBucketCreate: false,
+		BucketClaimOwner:  r.BucketName,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var accessKeys nb.S3AccessKeys
+	// if we didn't get the access keys in the create_account reply we might be talking to an older noobaa version (prior to 5.1)
+	// in that case try to get it using read account
+	if len(accountInfo.AccessKeys) == 0 {
+		log.Info("CreateAccountAPI did not return access keys. calling ReadAccountAPI to get keys..")
+		readAccountReply, err := r.SysClient.NBClient.ReadAccountAPI(nb.ReadAccountParams{Email: r.AccountName})
+		if err != nil {
+			return nil, err
+		}
+		accessKeys = readAccountReply.AccessKeys[0]
+	} else {
+		accessKeys = accountInfo.AccessKeys[0]
+	}
+
+	log.Infof("✅ Successfully created account %q with access to bucket %q", r.AccountName, r.BucketName)
+	return &accessKeys, nil
+}
+
+// DeleteAccount deletes the bar account
+func (r *APIRequest) DeleteAccount() error {
+
+	log := r.Provisioner.Logger
+	log.Infof("deleting account %q", r.AccountName)
+	err := r.SysClient.NBClient.DeleteAccountAPI(nb.DeleteAccountParams{Email: r.AccountName})
+
+	if err != nil {
+		if nbErr, ok := err.(*nb.RPCError); ok && nbErr.RPCCode == "NO_SUCH_ACCOUNT" {
+			log.Warnf("Account to delete was not found %q", r.AccountName)
+		} else {
+			return fmt.Errorf("failed to delete account %q. got error: %v", r.AccountName, err)
+		}
+	} else {
+		log.Infof("✅ Successfully deleted account %q", r.AccountName)
+	}
+
+	return nil
+}
+
+// DeleteBucket deletes the cosi bucket **including data**
+func (r *APIRequest) DeleteBucket() error {
+	// TODO we should support the different delete options - not completely closed by the KEP
+	var err error
+	log := r.Provisioner.Logger
+	info, err := r.SysClient.NBClient.ReadBucketAPI(nb.ReadBucketParams{Name: r.BucketName})
+	if nbErr, ok := err.(*nb.RPCError); ok && nbErr.RPCCode == "NO_SUCH_BUCKET" {
+		log.Warnf("Bucket to delete was not found %q", r.BucketName)
+		return nil
+	}
+	log.Infof("deleting bucket %q", r.BucketName)
+	if info.Namespace != nil {
+		err = r.SysClient.NBClient.DeleteBucketAPI(nb.DeleteBucketParams{Name: r.BucketName})
+	} else {
+		err = r.SysClient.NBClient.DeleteBucketAndObjectsAPI(nb.DeleteBucketParams{Name: r.BucketName})
+	}
+	if err != nil {
+		return fmt.Errorf("failed to delete bucket %q. got error: %v", r.BucketName, err)
+	}
+
+	log.Infof("✅ Successfully deleted bucket %q", r.BucketName)
+	return nil
+}
+
+func fetchUserCredentials(accessKeys nb.S3AccessKeys) map[string]*cosi.CredentialDetails {
+	s3Keys := make(map[string]string)
+	s3Keys["accessKeyID"] = accessKeys.AccessKey
+	s3Keys["accessSecretKey"] = accessKeys.SecretKey
+	creds := &cosi.CredentialDetails{
+		Secrets: s3Keys,
+	}
+	credDetails := make(map[string]*cosi.CredentialDetails)
+	credDetails["s3"] = creds
+	return credDetails
+}
+
+// prepareReplicationParams validates and prepare the replication params
```

Comment:
As discussed let's try to move these two functions which are also shared with OBC to somewhere else both can use (like util maybe)

Reply:
util can't be used as it will create an import cycle, 
so what I did it to import obc package in cosi and use the same function inorder to avoid double logic